### ISSUE TITLE
Array relations

### ISSUE THREAD

**Issue by kelloggm** [kelloggm/checker-framework#43]
Consider the following code:

```
    	int[] firstHalf = new int[array.length/2];
    	for(int i = 0; i < firstHalf.length; i++){
    		firstHalf[i] = array[i];
    	}
```

The UBC warns about the access to array. This is clearly a false positive, since i is `@LTLengthOf("firstHalf")`, and we can reason that firstHalf's MinLen has to be smaller than array's. But the UBC warns anyway. This showed up in a pilot of the IC study.

**Comment by kelloggm** [comment]
There's a testcase in 28c04c010ab34c6b191387442c0d51b9bd3f2524: Pilot2HalfLength.java

**Comment by ovshake** [comment]
If the length of array is 1 then this code shows ```java.lang.ArrayIndexOutOfBoundsException: 0``` error without the Index Checker. 
Maybe that is why this is showing positive. For eg:
```
   int[] array = {1};
   int[] firstHalf = new int[array.length/2];		
   for(int i = 0; i < firstHalf.length; i++){
    	firstHalf[i] = array[i];
    }
    System.out.print(firstHalf[0]+" ");
```

**Comment by jabbar86** [comment]
public class Demo {
	private static int[] getFirstHalf(int[] array) {
		         int[] firstHalf = new int[array.length];
		        for (int i = 0; i < firstHalf.length; i++) {
		             firstHalf[i] = array[i];
		         }
		         return firstHalf;
		     }
    public static void main(String[] args) {
    	int[] array={1,2};	
    	System.out.println(getFirstHalf(array));
    	        // System.out.println(firstHalf.length);
    }
}

I dont know whether it is correct or not but while running the program and checking using index checker it doesnt give any error

**Comment by kelloggm** [comment]
@ovshake, the only part about the code you included in your comment that can throw an `IndexOutOfBoundException` is this line: `System.out.print(firstHalf[0]+" ");`, which is not part of this issue.

@jabbar86 I ran the program you cited through the Index Checker, and it does report an error on this line: `firstHalf[i] = array[i];`, as this issue claims.

**Comment by kelloggm** [comment]
Closed in favor of #158
